xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -1992,3 +1992,48 @@ document.observe("xwiki:dom:loaded", function() {
   interceptXMLHttpRequest();
   interceptFetch();
 })();
+
+/**
+ * Initializes the submit-once protection on the current page. Search for all form elements with a 'xwiki-submit-once'
+ * class on the document (or on an updated sub-part of the dom). For each form, once the form is submitted, disable all
+ * submit buttons. Note that currently we are expecting the form to be submitted without client side validation. It is
+ * not advised to add the xwiki-submit-once class on forms with client side validation. Also, note that the value of
+ * the disabled buttons will not be sent server side. Therefore, it is not advised to use the 'xwiki-submit-once' class
+ * when this value is required server side.
+ *
+ *  @since 14.10.18
+ *  @since 15.5.3
+ *  @since 15.8
+ */
+require(['jquery'], ($) => {
+  /**
+   * Initializes the root element by adding event listeners to all forms with the class 'xwiki-submit-once'.
+   * When a form is submitted, re-submission is prevented by disabling all submit buttons.
+   *
+   * @param {Element} rootElement - The root element where the forms are located.
+   */
+  function init(rootElement) {
+    Array.from(rootElement.getElementsByClassName('xwiki-submit-once')).forEach(form => {
+      // Use a submit event handler with namespace 'xwiki-submit-once'. Before defining a handler, any potentially 
+      // registered listener is removed to prevent double registration of the same handler.
+      $(form).off("submit.xwiki-submit-once").on("submit.xwiki-submit-once", (e) => {
+        const submittingClass = 'xwiki-form-submitting';
+        // Prevent if already submitting.
+        if (form.classList.contains(submittingClass)) {
+          e.preventDefault();
```

Comment:
To me this logic is superfluous: we're marking the form as submitting as soon as a button is clicked so that we can then catch any other submit that may happen to stop it because it's already submitting.

This is what button disabling is supposed to handle / replace: the possibility for the form to be submitted after it has been submitted once, so all this logic here is completely unnnecessary (unless I missed something).

Reply:
We _may_ keep the xwiki-form-submitting class, if we really want to, but just as a tool to give to devs to make styling, not more.